src/platform/datapath_raw_socket.c

```diff
@@ -366,7 +366,7 @@ CxPlatDpRawParseIPv6(
     } VersionClassEcnFlow;
     VersionClassEcnFlow.Value = CxPlatByteSwapUint32(IP->VersionClassEcnFlow);
 
-    Packet->TypeOfService = (uint8_t)VersionClassEcnFlow.EcnField;
+    Packet->TypeOfService = ((uint8_t)VersionClassEcnFlow.EcnField) | (uint8_t)(VersionClassEcnFlow.Class << 2);
```

Comment:
I wonder if it makes a difference (in terms of instructions) to just cast the result of the OR to a `uint8_t` instead of casting both first...

Reply:
GCC outputs identical code for both versions; MSVC outputs the same number of instructions (but slightly different ordering); Clang outputs different instructions and the single cast has 2 fewer instructions
see here for my experiment https://godbolt.org/z/YW5o75cv3